### PR TITLE
Document version history

### PR DESCRIPTION
**Pull request by laurakalbag**
This PR adds a page documenting the version history feature under `Using the Studio > Edit > Version history`.

CleanShot 2023-03-24 at 12 20 06@2x

There’s no images in there yet, but there’s placeholder comments where I want to:
- [x]  add images

Other work to be done pending feedback:
- [x] Clarify which machines have an initial version
- [x] Clarify how version history works for other people’s/public machines

There’s a few other features bundled into this PR:
- We now have admonitions for `:::new` and `:::video`. (All admonitions are viewable on the example page at /docs/admonition-examples.)
CleanShot 2023-03-24 at 12 23 07@2x
CleanShot 2023-03-24 at 12 23 24@2x
- Admonitions now use Lucide icons so they fit better with the Studio.
- The copy for free trials/pro plan has been tweaked to be more readable on multiple pages.
- `Context` is now a top-level page inside the `XState` category, rather than being inside the `Actions` category.
- I’ve moved some of the video admonitions below page intros, because the intros should be at the top, and this is also what will show in search engine results/social previews etc.
- I’ve now documented that you can set and update the project visibility when creating a project and from Project Settings.

### CODE REVIEW

File: docs/versions.mdx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+## Team roles and versions
+
+Team viewers can only create new versions of machines in their team’s projects. Team owners, admins, and editors can view, create, restore, and delete versions of a team project’s machine.
+
+| Capability                               | Owner  | Admin  | Editor | Viewer |
+| ---------------------------------------- | ------ | ------ | ------ | ------ |
+| View team machine versions               | ✅ Yes | ✅ Yes | ✅ Yes | ✅ Yes |
+| Create new versions of team machines     | ✅ Yes | ✅ Yes | ✅ Yes | ✅ Yes |
+| Restore versions of team machines        | ✅ Yes | ✅ Yes | ✅ Yes | 🚫 No  |
+| Delete versions of team machines         | ✅ Yes | ✅ Yes | ✅ Yes | 🚫 No  |
```
Comment (kevinmaes) [inline docs/versions.mdx:116]:
```suggestion
| Delete versions of team machines         | ✅ Yes | ✅ Yes | 🚫 No | 🚫 No  |
```

File: docs/versions.mdx [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+1. Open the **Version history** panel from the right tool menu.
+2. Select the version you want to delete. The version will be previewed in the Editor.
+3. Use the delete icon button in the top bar to delete that version.
+4. The current version will then be opened in the Editor.
+
+<!-- TODO: ![A version previewed in the Editor with an arrow pointing to the Delete button.]() -->
+
+## Team roles and versions
+
+Team viewers can only create new versions of machines in their team’s projects. Team owners, admins, and editors can view, create, restore, and delete versions of a team project’s machine.
```
Comment (kevinmaes) [inline docs/versions.mdx:109]:
```suggestion
All team members can view past versions of machines in their team’s projects. Team owners, admins, and editors can also create and restore versions. Only owners and admins can delete versions.
```

**Review by kevinmaes — APPROVED** [review]
This is nearly perfect as is. Just 2 corrections I found so far.

There may be some upcoming changes to the way we handle initial versions and auto-saved versions but when they happen the instructions will be even simpler so no need to do anything now.

**Review by kevinmaes — CHANGES_REQUESTED** [review]
I was so eager to hit Approve I forgot that I **am** actually requesting changes or at least accepting the 2 suggestions I made. Not sure if I can revise my review here now.

**Comment by laurakalbag** [comment]
https://linear.app/statelyai/issue/STA-3477

**Review by kevinmaes — APPROVED** [review]

**Review by kevinmaes — CHANGES_REQUESTED** [review]
Proposing a change to the "Initial version" section.

File: docs/versions.mdx [hunk 3]
```diff
@@ -44,7 +44,7 @@ Versions are named with their creation time by default and have the version crea
 
 ### Initial versions
 
-An initial version is created when a new machine is created or imported. Autosaved machines have a Stately Bot icon. For newer machines, the initial version is named “Initial version.” The initial version’s name for older machines is a timestamp of when the machine was created or imported. Machines created in 2022 may not have an initial version as they were made before the version history existed.
+An initial version is created when a new machine is created or imported. Autosaved machines are is “Initial version” and have a Stately Bot icon.
```
[the remark this answers is not in the document]
Comment (laurakalbag) [reply docs/versions.mdx]: @kevinmaes I used your copy but changed it into a list for readability.

File: docs/versions.mdx [hunk 4]
```diff
@@ -44,7 +44,13 @@ Versions are named with their creation time by default and have the version crea
 
 ### Initial versions
 
-An initial version is created when a new machine is created or imported. Autosaved machines are is “Initial version” and have a Stately Bot icon.
+An initial version is automatically made for new machines created when:
+
+- you import a machine from code
+- you fork a machine from an existing machine or project
+- you duplicate an existing machine or project
```
Comment (kevinmaes) [inline docs/versions.mdx:51]: Might be more accurate to say "you duplicate an existing machine in its project" 

I don't think we "duplicate" projects yet although that could be cool, maybe.

File: docs/versions.mdx [hunk 5]
```diff
@@ -48,7 +48,7 @@ An initial version is automatically made for new machines created when:
 
 - you import a machine from code
```
Comment (kevinmaes) [inline docs/versions.mdx:49]: @laurakalbag We need to remove this one for now. We had trouble with the layout so we don't create an initial version anymore for import from code.

Comment (laurakalbag) [reply docs/versions.mdx]: @kevinmaes done!

**Review by kevinmaes — APPROVED** [review]
@laurakalbag These docs look great and we can release these as is but will soon need to follow it up by removing mention of the auto-saved versions when restoring a version since that should get released soon too.

**Comment by laurakalbag** [comment]
> @laurakalbag will soon need to follow it up by removing mention of the auto-saved versions when restoring a version since that should get released soon too.

@kevinmaes sorry I’ve not been keeping up on the version history engineering chat… we’ll no longer be autosaving versions when restoring?

**Comment by kevinmaes** [comment]
> > @laurakalbag will soon need to follow it up by removing mention of the auto-saved versions when restoring a version since that should get released soon too.
> 
> @kevinmaes sorry I’ve not been keeping up on the version history engineering chat… we’ll no longer be autosaving versions when restoring?

That's correct but the code is in Dev and not yet released. But it should be soon. I guess we can just remove that mention here and then release these.

**Pull request merged**